### ISSUE TITLE
nextcloud: overzealous sts header usage

### ISSUE THREAD

**Issue by arnottcr** [NixOS/nixpkgs#169465]
The `Strict-Transport-Security` header should only be added if [services.nextcloud.https](https://github.com/NixOS/nixpkgs/blob/nixos-21.11/nixos/modules/services/web-apps/nextcloud.nix#L148) is enabled, right?

https://github.com/NixOS/nixpkgs/blob/147b03fa8ebf9d5d5f6784f87dc61f0e7beee911/nixos/modules/services/web-apps/nextcloud.nix#L910

**Comment by aidalgol** [comment]
I don't think so.  The description for that option says "Use https for generated links."  The commit you linked is from last Feburary (which may be the version in NixOS 21.11 – I'm not sure), and on `master`, that header is controlled by the boolean option `services.nextcloud.nginx.recommendedHttpHeaders`.

The upstream project recommendes this header, as far as I can tell, and the Nextcloud admin panel warns you if this is not set, so I think it is semantically correct for this header to be added if the "recommended HTTP headers" flag is set, but it does get in the way if you are trying to serve Nextcloud as a Tor .onion service, where you don't want to be using SSL/TLS.

**Comment by arnottcr** [comment]
Good to hear that is different in master, but it is not just onion services, many internal services lack tls because the vpn or tls termination is done upstream and nextcloud only speaks http. I would expect we want a single boolean to turn on/off tls, no?

**Comment by aidalgol** [comment]
I had not thought of that use case.  I would say that is a good reason to have this controlled by the same flag.
